modules/blocks/process.js

```diff
@@ -475,7 +475,14 @@ Process.prototype.loadBlocksFromPeer = function(peer, cb) {
 					return setImmediate(eachSeriesCb);
 				}
 				// ...then process block
-				return processBlock(block, eachSeriesCb);
+				return processBlock(block, err => {
+					// Ban a peer if block validation fails
+					// Invalid peers won't get chosen in the next sync attempt
+					if (err) {
+						library.logic.peers.ban(peer);
```

Comment:
Is there a reason why we remove peer on here: https://github.com/LiskHQ/lisk/pull/2253/files#diff-60e1e043f6f3fdde809b81641313508bR447 but only ban peer here?